std/algorithm/iteration.d

```diff
@@ -836,7 +838,7 @@ template each(alias pred = "a")
 
     enum isForeachBinaryIterable(R) =
         is(typeof((R r) {
-            foreach (i, ref a; r)
+            foreach (ref i, ref a; r)
```

Comment:
Is this done for performance?
It seems dangerous to me.
Can you add a test that shows the user can't accidentally modify the loop variable?